parts/linux/cloud-init/artifacts/components.json

```diff
@@ -8,412 +10,26 @@
-      "multiArchVersions": []
-    },
-    {
-      "downloadURL": "mcr.microsoft.com/oss/virtual-kubelet/virtual-kubelet:*",
-      "amd64OnlyVersions": [],
-      "multiArchVersions": []
-    },
-    {
-      "downloadURL": "mcr.microsoft.com/azure-policy/policy-kubernetes-addon-prod:*",
-      "amd64OnlyVersions": [],
-      "multiArchVersions": [
-        "1.7.0",
-        "1.5.0"
-      ]
-    },
-    {
-      "downloadURL": "mcr.microsoft.com/azure-policy/policy-kubernetes-webhook:*",
-      "amd64OnlyVersions": [],
-      "multiArchVersions": [
-        "1.7.0",
-        "1.5.0"
-      ]
-    },
-    {
-      "downloadURL": "mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:*",
-      "amd64OnlyVersions": [],
-      "multiArchVersions": []
-    },
-    {
-      "downloadURL": "mcr.microsoft.com/aks/aks-node-ca-watcher:*",
-      "amd64OnlyVersions": [],
-      "multiArchVersions": [
-        "master.240530.2"
-      ]
-    },
-    {
-      "downloadURL": "mcr.microsoft.com/oss/kubernetes/windows-gmsa-webhook:*",
-      "amd64OnlyVersions": [],
-      "multiArchVersions": [
-        "v0.7.1"
-      ]
-    },
-    {
-      "downloadURL": "mcr.microsoft.com/oss/kubernetes/kube-proxy:*",
-      "amd64OnlyVersions": [],
-      "multiArchVersions": [
-        "v1.27.13",
-        "v1.27.13-2",
-        "v1.27.14-1",
-        "v1.27.15",
-        "v1.27.16",
-        "v1.28.10-1",
-        "v1.28.9",
-        "v1.28.9-1",
-        "v1.28.11",
-        "v1.28.12",
-        "v1.29.4",
-        "v1.29.4-1",
-        "v1.29.5-1",
-        "v1.29.6",
-        "v1.29.7",
-        "v1.30.1-1",
-        "v1.30.2-hotfix.20240613",
-        "v1.30.3"
+        "cuda-550.90.07-sha-b40b85"
       ]
     }
-  ],
+    ],
   "Packages": [
-    {
-      "name": "oras",
-      "downloadLocation": "/usr/local/bin",
-      "downloadURIs": {
-        "default": {
-          "current": {
-            "versions": [
-              "1.2.0"
-            ],
-            "downloadURL": "https://github.com/oras-project/oras/releases/download/v${version}/oras_${version}_linux_${CPU_ARCH}.tar.gz"
-          }
-        }
-      }
-    },
-    {
-      "name": "runc",
-      "downloadLocation": "/opt/runc/downloads",
-      "downloadURIs": {
-        "ubuntu": {
-          "r1804": {
-            "versions": [
-              "1.1.12-ubuntu18.04u1"
-            ]
-          },
-          "r2004": {
-            "versions": [
-              "1.1.12-ubuntu20.04u1"
-            ]
-          },
-          "r2204": {
-            "versions": [
-              "1.1.12-ubuntu22.04u1"
-            ]
-          }
-        },
-        "mariner": {
-          "current": {
-            "versions": [
-              "1.1.9-5.cm2"
-            ]
-          }
-        }
-      }
-    },
     {
       "name": "containerd",
+      "type": "OCI |Tar|tgz|custom",
       "downloadLocation": "/opt/containerd/downloads",
+      "deps": ["runc", "<another package>"],
```

Comment:
Adding a `deps` (optional) here so that we can ensure the dependencies are installed before this package. Now we are relying on the order of the packages. But having a `deps` should be more reliable. Need additional implementation in the codes for sure.